From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Phoenix616 <mail@moep.tv>
Date: Tue, 5 Mar 2019 23:57:38 -0500
Subject: [PATCH] Improve damage handling of dead entities

Manual application of 73597ed865e92dbfad404b69eb6468fb7e129ad2 until the
upstream patch file is sorted out.

diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index b4f21171..77ea0ec4 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -0,0 +0,0 @@ public abstract class EntityLiving extends Entity {
             return false;
         } else if (this.world.isClientSide) {
             return false;
-        } else if (this.getHealth() <= 0.0F) {
+        } else if (this.dead || this.killed || this.getHealth() <= 0.0F) { // Paper // CraftBukkit - Don't allow entities that got set to dead/killed elsewhere to get damaged and die
             return false;
         } else if (damagesource.p() && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
             return false;
--